commands/key_add.go

```diff
@@ -0,0 +1,89 @@
+package commands
+
+import (
+	"fmt"
+
+	"github.com/MichaelMure/git-bug/cache"
+	"github.com/MichaelMure/git-bug/commands/select"
+	"github.com/MichaelMure/git-bug/identity"
+	"github.com/MichaelMure/git-bug/input"
+	"github.com/MichaelMure/git-bug/util/interrupt"
+	"github.com/spf13/cobra"
+)
+
+var (
+	keyAddArmoredFile    string
+	keyAddArmored        string
+	keyAddExpireUnixTime int64
+)
+
+func runKeyAdd(cmd *cobra.Command, args []string) error {
+	backend, err := cache.NewRepoCache(repo)
+	if err != nil {
+		return err
+	}
+	defer backend.Close()
+	interrupt.RegisterCleaner(backend.Close)
+
+	id, args, err := _select.ResolveUser(backend, args)
+	if err != nil {
+		return err
+	}
+
+	if len(args) > 0 {
+		return fmt.Errorf("unexpected arguments: %s", args)
+	}
+
+	if keyAddArmoredFile != "" && keyAddArmored == "" {
+		keyAddArmored, err = input.BugCommentFileInput(keyAddArmoredFile)
+		if err != nil {
+			return err
+		}
+	}
+
+	if keyAddArmoredFile == "" && keyAddArmored == "" {
+		keyAddArmored, err = input.IdentityVersionKeyEditorInput(backend, "")
```

Comment:
what about generating a new key instead ? Ideally using git-bug, even with signature, should not require for the user to learn PGP or even knowing what that is. Power user should still be allowed to input their own keys but that's their choice to go the hard way.

Reply:
Generating a new key means the private key has to be stored in a file, and to be used with `gpg` it has to be imported. Can be done, but we can live without it.

Reply:
Storing the private key in the git config, even is it's not perfect seems like an OK solution to me. It would be nice to be entirely independent of a `gpg` binary, but I don't know if that's possible without migrating to go-git.

Reply:
Can be done, but can also be added later?

Reply:
Well, I'm not going to refuse this useful contribution because it doesn't meet 100% of what I think should happen but I'm considering that a requirement for an actual release.

I'm also saying that because it *looks like* that you are going on a more complicated path but I may be wrong.

Reply:
We do need `gpg` and the private key to be in the gpg keyring to be able to create signed commits with `git`.

Reply:
I see. I'll be happy when I can ditch the git CLI exec repo, it cause so many problems :(

Reply:
I mean the private key will be used also to create the normal commits in the repository, so `gpg` will be used since creating those commits will be done with `git`, not with anything git-bug or go-git.